include/mxnet/tensor_blob.h

```diff
@@ -306,6 +288,16 @@ class TBlob {
         this->shape_.FlatTo3D(axis_begin, axis_end), stream);
   }
 
+  inline void SetDLTensor(int dev_mask, int dev_id) {
```

Comment:
I'm not sure whether we should make this public or let NDArray access this by friend method